products/bigquery/api.yaml

```diff
@@ -383,6 +383,613 @@ objects:
               A-Z), numbers (0-9), or underscores (_). The maximum length
               is 1,024 characters.
             required: true
+  - !ruby/object:Api::Resource
+    name: 'Job'
+    kind: 'bigquery#job'
+    base_url: projects/{{project}}/jobs
+    delete_url: projects/{{project}}/jobs/{{job_id}}/cancel
+    delete_verb: :POST
+    self_link: projects/{{project}}/jobs/{{job_id}}
+    input: true
+    description: |
+      Jobs are actions that BigQuery runs on your behalf to load data, export data, query data, or copy data.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'etag'
+        description: |
+          A hash of the resource.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'id'
+        output: true
+        description: |
+          Opaque ID field of the job.
+      - !ruby/object:Api::Type::String
+        name: 'user_email'
+        output: true
+        description: |
+          Email address of the user who ran the job.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'configuration'
+        description: 'Describes the job configuration.'
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'jobType'
+            description: |
+              The type of the job.
+            output: true
+          - !ruby/object:Api::Type::String
+            name: 'jobTimeoutMs'
+            description: |
+              Job timeout in milliseconds. If this time limit is exceeded, BigQuery may attempt to terminate the job.
+          - !ruby/object:Api::Type::KeyValuePairs
+            name: 'labels'
+            description: |
+              The labels associated with this job. You can use these to organize and group your jobs.
+          - !ruby/object:Api::Type::NestedObject
+            name: 'query'
+            description: 'Configures a query job.'
+            exactly_one_of:
+              - query
```

Comment:
Not here.  It's a little confusing how this is implemented, but with `exactly_one_of` and `at_least_one_of`, the list needs to be exactly how it will be in the downstream code.  I flattened `configuration` in `terraform.yaml`, so this will show up as just `query`, etc downstream.